src/RequestFactory.php

```diff
@@ -246,4 +302,73 @@ private function isJson(array $headers)
         // http://greenbytes.de/tech/webdav/rfc2616.html#rfc.section.14.17
         return strpos($headers['Content-Type'], 'application/json') === 0;
     }
+
+    /**
+     * @param array $headers
+     */
+    private function headerIsJsonOrFail(array $headers) {
+        if (!$this->isJson($headers)) {
+            throw new  RuntimeException("Content-Type header is not equal to \'application/json'");
+        }
+    }
+
+    /**
+     * @param string $uri
+     * @param array  $request
+     * @return array
+     */
+    private function getPlaceholdersFromUriAndHeader($uri, $request) {
+        $result = [];
+        $placeholders = $this->getPlaceholdersInString($uri);
+        $placeholders[] = $this->getPlaceholdersInArray($request['headers']);
+
+        array_walk_recursive($placeholders,function($v, $k) use (&$result){ $result[] = $v; });
```

Comment:
If you want to get just the values of an array there is a function called array_values